### PR TITLE
Add QR code ticketing system

### PR DESCRIPTION
**Pull request by BillySBarrett**
Co-authored-by: Henrik Skog <henrikskog@users.noreply.github.com>

## Changelog

* Add endpoints for handeling qr code registration
* Give warning when user is already registered
* Show modal for user registered
* Add button to create qr code on event
* This is only MVP, encryption and possible Apple Wallet integrations can come later

### CODE REVIEW

**Comment by junlarsen** [comment]
:rocket:

**Review by henrikskog — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/app/(dashboard)/event/[id]/attendees-page.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ interface Props {
 }
 
 const Page: FC<Props> = ({ attendance }) => {
-  const { attendees } = useEventAttendeesGetQuery(attendance.id)
+  const [result, setResult] = useState("")
```
[the remark this answers is not in the document]
Comment (BillySBarrett) [reply apps/dashboard/src/app/(dashboard)/event/[id]/attendees-page.tsx]: This is removed now

File: apps/web/src/app/events/components/TicketBox.tsx
[the remark this answers is not in the document]
Comment (BillySBarrett) [reply apps/web/src/app/events/components/TicketBox.tsx:1]: I agree, this is updated

File: apps/web/src/app/events/[id]/page.tsx [hunk 2]
```diff
@@ -90,6 +87,7 @@ const EventDetailWithAttendancePage: FC<WithAttendanceProps> = ({ user, attendan
         <EventInfoBox event={event} />
         <div className="flex flex-1 flex-col">
           <AttendanceBox sessionUser={user} attendance={attendance} pools={pools} event={event} />
+          <TicketBox userid={user.id} />
```
[the remark this answers is not in the document]
Comment (BillySBarrett) [reply apps/web/src/app/events/[id]/page.tsx]: I thought this was already implemented but its now fixed

**Review by henrikskog — APPROVED** [review]
lint then this looks good!

File: packages/core/src/modules/attendance/attendee-service.ts [hunk 3]
```diff
@@ -29,6 +30,10 @@ export interface AttendeeService {
   getByAttendanceId(attendanceId: string): Promise<AttendeeUser[]>
   getByAttendancePoolId(id: AttendancePoolId): Promise<AttendeeUser[]>
   updateAttended(attended: boolean, id: AttendeeId): Promise<Attendee>
+  handleQrCodeRegistration(
+    userId: UserId,
+    attendanceId: AttendanceId
+  ): Promise<{ attendee: Attendee; user: User; alreadyAttended: boolean }>
```
Comment (junlarsen) [inline packages/core/src/modules/attendance/attendee-service.ts:36]: I think it's easier to read if you move the return type into an interface/type

Comment (BillySBarrett) [reply packages/core/src/modules/attendance/attendee-service.ts]: I agree, it is now changed to QrCodeRegistrationAttendee type

**Pull request merged**